usl_models/usl_models/flood_ml/metastore.py

```diff
@@ -1,8 +1,42 @@
+import datetime
 from typing import Any, Sequence, TypeVar
 import urllib.parse
 
 from google.cloud import firestore  # type:ignore[attr-defined]
 
+import usl_models.flood_ml.model_params
+
+
+def write_model_metadata(
+    db: firestore.Client,
+    gcs_model_dir: str,
+    sim_names: Sequence[str],
+    model_params: usl_models.flood_ml.model_params.FloodModelParams,
+) -> str:
+    """Writes information on a trained model to the metastore.
+
+    Args:
+      db: The firestore client to use when retrieving metadata.
+      gcs_model_dir: The location of the saved model in GCS.
+      sim_names: The simulations on which the model was trained.
+      model_params: The parameters used to train the model.
+
+    Returns:
+      The document ID of the written model metadata document.
+    """
+    # Use the GCS location of the model as a unique value for the ID.
+    # URL-escape the ID, as characters such as / are not allowed in document IDs.
+    model_id = urllib.parse.quote(gcs_model_dir, safe=())
+    db.collection("models").document(model_id).set(
+        {
+            "trained_on": [_get_simulation_doc(db, sim_name) for sim_name in sim_names],
+            "trained_at_utc": datetime.datetime.utcnow(),
```

Comment:
What's the value of this? Is this mostly to track when we saved a model?

Reply:
yeah, to help distinguish the different experiments -- a better marker would be something like a commit hash or a version of the code trained on

Reply:
What about adding something like an experiment name or description? So the trainer task could (optionally) take in a short descriptor and include that when we save the model?

Reply:
now with model name!